File: 000. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            Integer index = map.get(target - nums[i]);
            if (index != null) {
                return new int[] {index, i};  
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: 000. Two Sum/Solution.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>solutionMap;
        
        for (int i = 0; i < nums.size(); i++) {
            auto it = solutionMap.find(target - nums[i]);
            if(it != solutionMap.end()) {
                return vector<int> {it->second, i};
            } 
            solutionMap[nums[i]] = i;
        }
    }
};
